src/core/net/srp_server.cpp

```diff
@@ -1133,33 +1139,30 @@ void Server::HandleUpdate(const Dns::UpdateHeader &aDnsHeader, Host &aHost, cons
 
         if (aHost.FindService(service.GetServiceName(), service.GetInstanceName()) == nullptr)
         {
-            Service *newService =
-                aHost.AddNewService(service.GetServiceName(), service.GetInstanceName(), service.IsSubType());
+            Service *newService = aHost.AddNewService(service.GetServiceName(), service.GetInstanceName(),
+                                                      service.IsSubType(), aMetadata.mRxTime);
 
             VerifyOrExit(newService != nullptr, error = kErrorNoBufs);
-            newService->mDescription.mTimeLastUpdate = TimerMilli::GetNow();
-            newService->mIsDeleted                   = true;
+            newService->mDescription.mUpdateTime = aMetadata.mRxTime;
+            newService->mIsDeleted               = true;
         }
     }
 
 exit:
-    if (error != kErrorNone)
+    if ((error == kErrorNone) && aMetadata.IsDirectRxFromClient() && (mServiceUpdateHandler != nullptr))
```

Comment:
I don't think we can. For now I limited it in the code to the "IsDirectRxFromClient" in this PR.

I had a chat on this with @jwhui as well. Here is quick summary of some of the thoughts
- I don't think we can publish (for example on mDNS) the same service from multiple SRP server or it may create conflicts (?)
- If we want this behavior, we can have the original SRP server (the first SRPL server which gets the message from the client directly) to do this.
   - Though this model may be tricky to realize since a client may initially send its SRP update to one server and then later send it to another (this is actually encouraged by anycast address model and SRPL key point is to make all SRP servers be the same to the clients).
- One idea is we only do a name check (so that the name is not in use by another entity on infra side) only if it is a new name (not an update to a previous name) and  only do this on the SRP server which receives the message directly from client (and not on other SRPL partners).

Thoughts? @Abhayakara any ideas?

Reply:
It is possible to advertise the same names on mDNS and not defend them (i.e. intentionally allowing conflicts).

Reply:
If the data being advertised is the same, there's no conflict.  However, when the data changes, you get a conflict until nobody is advertising the stale data anymore. So what we do is to just try to advertise the new data; if we get a conflict, we try again shortly. When the last peer advertising the old data gets the update, it will advertise the new data and not get a conflict. So the effect should be that the update is visible to the network almost immediately, and then the other servers catch up quickly (we do exponential backoff just in case, but the first retry is at 15 seconds).

Also, however, see also https://datatracker.ietf.org/doc/html/draft-tllq-tsr-00.html, which fully addresses the problem.

Reply:
Thanks. The TSR RR is indeed a good solution to address this. 

I will change this PR for now to allow second (all) SRPL partners to also invoke the `mServiceUpdateHandler` (if the callback is set).

Reply:
We don't currently have retry mechanism implemented, so we should consider adding that.

I agree the TSR RR mechanism is a good solution.

Reply:
I agree, we may need a retry mechanism. I would say for now, we can assume that the callback  `mServiceUpdateHandler` implementation would handle this (retry if needed). I think later on we can re-think the `mServiceUpdateHandler` model and whether to add retry in `Srp::Server` code directly.